src/os/bluestore/BlueStore.cc

```diff
@@ -3720,7 +3720,7 @@ bool BlueStore::OmapIteratorImpl::valid()
 {
   RWLock::RLocker l(c->lock);
   bool r = o->onode.has_omap() && it && it->valid() &&
-    it->raw_key().second <= tail;
+    it->raw_key().second < tail && it->raw_key().second != tail;
```

Comment:
why adding "it->raw_key().second != tail"?  Isn't "it->raw_key.second < tail" enough?

Reply:
There are scenarios where an error occurs, such as `read_log_and_missing`，So here's what we did

Reply:
@cxytt - I simply mean that last part of this expression (it->raw_key()second != tail) is redundant. True for it->raw_key().second < tail automatically results in true for ... != tail.

Reply:
（─.─||）fixed